modules/ballerina-config/src/main/java/org/ballerinalang/config/ConfigProcessor.java

```diff
@@ -0,0 +1,163 @@
+            fileInstanceConfigs.forEach((key, val) -> resolvedInstanceConfigs.putIfAbsent(key, val));
+        }
+
+        // Merge the runtime configurations with the already resolved configs.
+        // Any configs already resolved and present are overridden.
+        resolvedGlobalConfigs.putAll(runtimeGlobalConfigs);
+        resolvedInstanceConfigs.forEach((key, val) -> {
+            if (runtimeInstanceConfigs.containsKey(key)) {
+                val.putAll(runtimeInstanceConfigs.get(key));
+            }
+        });
+        runtimeInstanceConfigs.forEach((key, val) -> resolvedInstanceConfigs.putIfAbsent(key, val));
+
+        configRegistry.setGlobalConfigs(resolvedGlobalConfigs);
+        configRegistry.setInstanceConfigs(resolvedInstanceConfigs);
+    }
+
+    private void lookUpVariables(Map<String, String> globalConfigs, Map<String, Map<String, String>> instanceConfigs) {
+        globalConfigs.keySet().forEach(key -> {
+            String value = System.getenv(convertToEnvVarFormat(key));
+            value = (value == null) ? System.getProperty(key) : value;
+
+            if (value != null) {
+                // replace the config value if there is an environment variable of the same name
+                resolvedGlobalConfigs.put(key, value);
+                globalConfigs.remove(key);
+            }
+        });
+
+        instanceConfigs.keySet().forEach(instanceId -> {
+            Map<String, String> configInstance = instanceConfigs.get(instanceId);
+            configInstance.keySet().forEach(key -> {
+                String value = System.getenv(convertToEnvVarFormat(key, instanceId));
+                value = (value == null) ? System.getProperty(key) : value;
+                if (value != null) {
+                    // replace the config value if there is an environment variable of the same name
+                    if (resolvedInstanceConfigs.containsKey(instanceId)) {
+                        resolvedInstanceConfigs.get(instanceId).put(key, value);
+                    } else {
+                        Map<String, String> map = new HashMap<>();
+                        map.put(key, value);
+                        resolvedInstanceConfigs.put(instanceId, map);
+                    }
+                    configInstance.remove(key);
+                }
+            });
+        });
+    }
+
+    private String convertToEnvVarFormat(String var) {
+        return var.toUpperCase(Locale.ROOT).replace('.', '_');
+    }
+
+    private String convertToEnvVarFormat(String var, String instanceId) {
+        return instanceId.toUpperCase(Locale.ROOT) + "__" + convertToEnvVarFormat(var);
+    }
+
+    private File getConfigFile(String fileLocation) {
+        File confFile;
+        if (fileLocation != null) {
+            confFile = new File(fileLocation);
+            if (!confFile.exists()) {
+                throw new BallerinaException("failed to parse ballerina.conf file: " + fileLocation);
```

Comment:
I think it would be better throw an error with a specific message similar to "file not found" etc. here, rather than a generic error.

Reply:
+1